ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/StateMachine.hs

```diff
@@ -1667,6 +1671,10 @@ traceEventName = \case
     TraceGCEvent                ev    -> "GC."                <> constrName ev
     TraceIteratorEvent          ev    -> "Iterator."          <> constrName ev
     TraceSnapshotEvent          ev    -> "Ledger."            <> constrName ev
+    -- This one is deprecated but unless we match it we get non-exhaustiveness
+    -- warnings. When this constructor is deleted we must remove this match as
+    -- well as re-enable the deprecation warnings at the top of this file.
+    TraceLedgerEvent            ev    -> "Ledger."            <> constrName ev
```

Comment:
Suggestion: raise an `error` in the LHS of the match to ensure matches don't go unnoticed?

Reply:
That would probably be the right thing to do, but in this case the code already compiles without this constructor, this is just to silence the warning of non-exhaustiveness. We will simple remove it eventually